packages/react-scripts/scripts/init.js

```diff
@@ -108,11 +115,18 @@ module.exports = function(
   );
   if (fs.existsSync(templateDependenciesPath)) {
     const templateDependencies = require(templateDependenciesPath).dependencies;
-    args = args.concat(
+    const args = baseArgs.concat(
       Object.keys(templateDependencies).map(key => {
         return `${key}@${templateDependencies[key]}`;
       })
     );
+
+    if (args.length > baseArgs.length) {
```

Comment:
Is a length check really what we want here? What if they remove `react` and `react-dom` and replace it with `name-your-dom-library`?

Can we check this by other means -- or just assume that if a template file exists, always install.

Reply:
`react` and `react-dom` installation is handled by the `create-react-app` package rather than `react-scripts` (https://github.com/callummr/create-react-app/blob/6489071ba587e346b5f3cc97d65fb37c09380d59/packages/create-react-app/createReactApp.js#L284), so the template deps don't affect those. It can contain either 0 or 1+ _extra_ dependencies, so I think a length check works?